### PR TITLE
Fixing broken `1.0.latest` CI

### PR DESCRIPTION
**Pull request by leahwicz**
### Description
We haven't been keeping up with CI for the `1.0.latest` branch. Some things have changed or needed updating that haven't gotten backporting so this PR takes care of that and fixes CI

Backports / Fixes (all of these exist on `1.1.latest` already so we are good there):
- Update CI to run the Bigquery `1.0.latest` branch against the Core `1.0.latest` branch. 
- Backport #196. 
- Backport #93 - This was done for Core's `1.0.latest` branch already so that's why it's failing here. 

### Checklist

- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-bigquery next" section.

### CODE REVIEW

**Comment by leahwicz** [comment]
@nathaniel-may as I was working on getting CI to run for release branches, I noticed that CI was broken for `1.0.latest`. I'm assuming that this will be the case for all the adapters so we should take a look and make sure we unblock those repo's release branches as well (i.e. `1.0.latest`)

**Review by McKnight-42 — APPROVED** [review]
Looks good to me

File: tests/unit/utils.py
Comment (nathaniel-may) [inline tests/unit/utils.py:256]: oh wow this is very much non obvious 🤦‍♂️

**Pull request merged**
